### PR TITLE
Streamline release process

### LINKED ISSUE #36: Automate release using github-release
See https://github.com/getcarina/carina/blob/master/release.sh for an example of using `github-release`.

### LINKED ISSUE #51: Download latest release not latest tag
There is an awkward period between when a tag is created and when the release is cut where anyone running the install will end up downloading 404 pages instead of the binaries in the (soon to be created) release.

### LINKED ISSUE #56: Release shell scripts to the releases page
Do this instead of relying on whatever is on master.

### PR DESCRIPTION
**Pull request by carolynvs**
- [x] Fixes #56 (Release shell scripts to the releases page)
- [x] Fixes #51 (Download latest release, not latest tag)
- [x] Fixes #36 (Automate release using github-release)
- [ ] Upload release artifacts to cloud files instead of github release page

**Build and Test Process**
Use `make` to build and output to `dvm-helper/dvm-helper`. Use `make cross-build` to build all platforms (then manually copy the desired binary from the repo-path/bin).

**Release Process**
1. `git checkout master`
2. `git pull`
3. `git tag 0.2.0 -a -m ""`.
4. `git push --follow-tags`
5. `./release.sh 0.2.0`

### CODE REVIEW

**Comment by carolynvs** [comment]
@rgbkrk Do you see any way to have a generic link to the latest install script? I think once we move to having the install scripts in the release files and not pointing the master, then we will need to update the readme on each release to point to the latest install script.

It doesn't look like GitHub provides a way to link to the latest release file, e.g. https://github.com/getcarina/dvm/releases/download/0.1.3/install.sh works but https://github.com/getcarina/dvm/releases/download/latest/install.sh doesn't.

**Comment by rgbkrk** [comment]
We can host the link as the tagged release copy, updating that with the release script - either in the README or...

We could set up a page for getcarina.com that contains a URL with latest that gets updated to be this tagged release underneath.

**Comment by carolynvs** [comment]
Now that download.getcarina.com is proxying to the latest release, the install scripts have been simplified and we can just use https://download.getcarina.com/getcarina/dvm/latest/install.sh, etc to download from the installer script.

@rgbkrk Can you take a look and let me know if this looks ok to merge?

**Comment by rgbkrk** [comment]
This certainly makes sense for the dvm case where we never have people download anything (other than install.sh) directly. In fact, it's pretty clear this is the simplest solution.

For `carina` I want the paths to be similar to get.docker.com or other download URLs for the sake of making it easy to download off of `/getcarina/carina/linux/amd64/latest/carina`. We'd have to stick with one strategy for naming (which is what we're doing here) or upload them to these paths (however we choose).

**Comment by carolynvs** [comment]
@rgbkrk What if the carina cli used an install script just like dvm, e.g. `curl http://path/to/install.sh | sh`? That certainly would be easier than the current instructions. Plus the file name wouldn't matter anymore as the script handles the renaming. There are enough steps that I think it would be a nice improvement.

**Comment by carolynvs** [comment]
@rgbkrk I'm not clear if is okay to merge into dvm or if I should hold off?

**Comment by rgbkrk** [comment]
URL examples for what we'll be using:

```
https://download.getcarina.com/dvm/$TAG/$( uname -s )/$( uname -m )/dvm-helper

Example:

https://download.getcarina.com/dvm/latest/Darwin/x86_64/dvm-helper

For the shell scripts (cmd, ps1, sh), they'll be at:

https://download.getcarina.com/dvm/$TAG/dvm.ps1
```

**Comment by rgbkrk** [comment]
```
22:16:00 rgbkrk@lack ~/.dvm$ rm -rf *
22:16:02 rgbkrk@lack ~/.dvm$ ls
22:16:03 rgbkrk@lack ~/.dvm$ DVM_DIR=$HOME/.dvm
22:16:10 rgbkrk@lack ~/.dvm$ export DVM_DIR=$HOME/.dvm
22:16:13 rgbkrk@lack ~/.dvm$ curl -s https://download.getcarina.com/dvm/latest/install.sh | sh
Downloading dvm.sh...
######################################################################## 100.0%
Downloading dvm-helper...
######################################################################## 100.0%

Docker Version Manager (dvm) has been installed to /Users/kyle6475/.dvm
Run the following command to start using dvm. Then add it to your bash profile (e.g. ~/.bashrc or ~/.bash_profile) to complete the installation.

	source /Users/kyle6475/.dvm/dvm.sh
22:16:19 rgbkrk@lack ~/.dvm$ ls
dvm-helper dvm.sh
22:16:20 rgbkrk@lack ~/.dvm$ source dvm.sh
22:16:27 rgbkrk@lack ~/.dvm$ dvm version
/Users/kyle6475/.dvm/dvm-helper/dvm-helper: line 1: syntax error near unexpected token `<'
/Users/kyle6475/.dvm/dvm-helper/dvm-helper: line 1: `<h1>404, head on back to <a href="https://getcarina.com">https://getcarina.com</a></h1>'
22:16:41 rgbkrk@lack ~/.dvm$ cat dvm-helper/dvm-helper
<h1>404, head on back to <a href="https://getcarina.com">https://getcarina.com</a></h1>
```

**Comment by rgbkrk** [comment]
I see what you mean about checksums not happening here.

**Comment by rgbkrk** [comment]
Direct download with `curl https://download.getcarina.com/dvm/latest/Darwin/x86_64/dvm-helper -o dvm-helper` worked though.

File: install.ps1 [hunk 1]
```diff
@@ -26,22 +26,20 @@ function downloadDvm([string] $dvmDir) {
   if( [System.Environment]::Is64BitOperatingSystem ) { $arch = "amd64" } else { $arch = "386"}
 
   # Download latest release
-  $latestTag = (ConvertFrom-Json (wget https://api.github.com/repos/getcarina/dvm/tags).Content) | select -ExpandProperty name | select -first 1
-
-  $webClient.DownloadFile("https://github.com/getcarina/dvm/releases/download/$latestTag/dvm-helper-windows-$arch.exe", "$tmpDir\dvm-helper-windows-$arch.exe")
-  $webClient.DownloadFile("https://github.com/getcarina/dvm/releases/download/$latestTag/dvm-helper-windows-$arch.exe.sha256", "$tmpDir\dvm-helper-windows-$arch.exe.256")
+  $webClient.DownloadFile("https://download.getcarina.com/dvm/latest/windows/$arch/dvm-helper.exe", "$tmpDir\dvm-helper.exe")
+  $webClient.DownloadFile("https://download.getcarina.com/dvm/latest/windows/$arch/dvm-helper.exe.sha256", "$tmpDir\dvm-helper.exe.256")
```
Comment (rgbkrk) [inline install.ps1]: These need to be using the `x86_64` version for the paths rather than the golang arch.

File: install.sh [hunk 2]
```diff
@@ -54,10 +54,9 @@ install_dvm_helper() {
   [ $(getconf LONG_BIT) == 64 ] && DVM_ARCH="amd64" || DVM_ARCH="386"
 
   # Download latest release
-  LATEST_TAG=$(curl -s https://api.github.com/repos/getcarina/dvm/tags | grep name -m 1 | awk '{print $2}' | cut -d '"' -f2)
   mkdir -p "$DVM_DIR/dvm-helper"
   bin="$DVM_DIR/dvm-helper/dvm-helper"
-  url=https://github.com/getcarina/dvm/releases/download/$LATEST_TAG/dvm-helper-$DVM_OS-$DVM_ARCH
+  url=https://download.getcarina.com/dvm/latest/$DVM_OS/$DVM_ARCH/dvm-helper
```
Comment (rgbkrk) [inline install.sh]: These need adjustment to be using `uname -s` for DVM_OS and `uname -m` for DVM_ARCH (replacing the `getconf` bits above).

Sample run that works for me:

```
$ curl https://download.getcarina.com/dvm/latest/$(uname -s)/$(uname -m)/dvm-helper -o dvmhelpah; chmod u+x dvmhelpah; ./dvmhelpah --version
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 7160k  100 7160k    0     0   293k      0  0:00:24  0:00:24 --:--:--  440k
Docker Version Manager version 0.1.4 (ca12e09)
```

**Comment by rgbkrk** [comment]
Looks like we just need to switch / add in the `uname` bits for the installer.

**Comment by rgbkrk** [comment]
Alright, made a PR for your PR that uses the new install paths: https://github.com/carolynvs/dvm/pull/1

**Comment by rgbkrk** [comment]
Install is working! This is great. It saddens me how big the go binary is compared to the old shell scripts. :cry: This will be much simpler to test though and I'm happy about the cross platform so there's that. :wink:

**Comment by rgbkrk** [comment]
Seeing an issue with `surge` though when trying to install over the top of an installation that isn't there

```
$ cat install.sh | sh
Downloading dvm.sh...
######################################################################## 100.0%
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
```

I'll bring this up with them [in chat tomorrow](https://twitter.com/surge_sh/status/662451073358032897).

File: install.ps1 (hunk 1 above)
Comment (rgbkrk) [inline install.ps1]: Done.

Comment (rgbkrk) [inline install.ps1]: This needs a capital `Windows`.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 $(GOBUILD) -o bin/Linux/x86_64/dvm-helper $(PACKAGE)
+	cd bin/Linux/x86_64 && shasum -a 256 dvm-helper > dvm-helper.sha256
+
+darwin: $(GOFILES)
+	CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GOBUILD) -o bin/Darwin/x86_64/dvm-helper $(PACKAGE)
+	cd bin/Darwin/x86_64 && shasum -a 256 dvm-helper > dvm-helper.sha256
+
+windows: $(GOFILES)
+	CGO_ENABLED=0 GOOS=windows GOARCH=amd64 $(GOBUILD) -o bin/Windows/x86_64/dvm-helper.exe $(PACKAGE)
+	cd bin/Windows/x86_64 && shasum -a 256 dvm-helper.exe > dvm-helper.exe.sha256
```
Comment (rgbkrk) [inline Makefile]: Cool, love that you changed up the Makefile for the new format as well.

**Comment by rgbkrk** [comment]
This is looking really great, just tried it out.

**Comment by rgbkrk** [comment]
Looks like installing over the top of a previous install is working perfectly too.

File: Dockerfile.build [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.5
+
+ADD . $GOPATH/src/github.com/getcarina/dvm/
+WORKDIR $GOPATH/src/github.com/getcarina/dvm/
+
+RUN make get-deps
```
Comment (rgbkrk) [inline Dockerfile.build]: :+1:

File: release.sh [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  make dvm-helper
+  echo "Out with the old, in with the new"
+  ./dvm-helper/dvm-helper --version
+  echo "---------------------------------"
+
+  # Build with the tag now for actual binary shipping
+  git fetch --tags $REMOTE
+  make build-tagged-for-release TAG=$TAG
+
+  echo "Creating a new relase on GitHub..."
```
Comment (rgbkrk) [inline release.sh]: We'll need to change these, but we can do it as a follow on.

**Pull request merged**